pkg/detectors/algoliaadminkey/algoliaadminkey.go

```diff
@@ -22,14 +24,14 @@ var (
 	client = common.SaneHttpClient()
 
 	// Make sure that your group is surrounded in boundary characters such as below to reduce false positives.
-	keyPat = regexp.MustCompile(detectors.PrefixRegex([]string{"algolia"}) + `\b([a-zA-Z0-9]{32})\b`)
-	idPat  = regexp.MustCompile(detectors.PrefixRegex([]string{"algolia"}) + `\b([A-Z0-9]{10})\b`)
+	keyPat = regexp.MustCompile(detectors.PrefixRegex([]string{"algolia", "docsearch", "apiKey"}) + `\b([a-zA-Z0-9]{32})\b`)
+	idPat  = regexp.MustCompile(detectors.PrefixRegex([]string{"algolia", "docsearch", "appId"}) + `\b([A-Z0-9]{10})\b`)
```

Comment:
@kashifkhan0771, regarding your comment from https://github.com/trufflesecurity/trufflehog/pull/3458#discussion_r1808314219:

> _I believe `appId` and `apiKey` is a very common term which can lead to false positives._

We're detecting these prefixes only in files that already contain the `algolia` or `docsearch` keyword, so I believe false positives would be very rare (and would be found through verification). Not including these terms might lead to false negatives, which —to my mind — is a bit more concerning.